src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +17,34 @@
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (IllegalArgumentException e) {
+            throw new DataProcessingException("The instance of Actor is not an entity.", e);
+        } catch (IllegalStateException | PersistenceException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save the actor: " + actor, e);
```

Comment:
The exception message could be more descriptive by including more context about the actor entity, such as its ID or name, to aid in debugging.